src/modules/ekf2/EKF/mag_fusion.cpp

```diff
@@ -211,11 +177,14 @@ bool Ekf::fuseMag(const Vector3f &mag, estimator_aid_source3d_s &aid_src_mag, bo
 		VectorState Kfusion = P * H / aid_src_mag.innovation_variance[index];
 
 		if (!update_all_states) {
-			for (unsigned row = 0; row <= 15; row++) {
+			unsigned mag_index_first = math::min(State::mag_I.idx, State::mag_B.idx);
```

Comment:
Alternatively we could also do it like here: https://github.com/PX4/PX4-Autopilot/blob/5f87f3a04652d74ea21f3d39ec687f287a7da45b/src/modules/ekf2/EKF/airspeed_fusion.cpp#L201-L203
to make sure we never assume that mag_I and mag_B are next to each other

Reply:
Yes that's probably better.

Reply:
updated